### DISCUSSION TITLE
Different variety of habits

### DISCUSSION THREAD

**Discussion by FOSS-Love** [iSoron/uhabits#891] [Feature Requests]
The loop habit tracker only supports on type of habit tracking technique. But i would like to have a few other types, for example :
- A bad habit breaker (quitting smoking)
  - auto-check can be helpful to acheieve this, so i can uncheck the day i relapse.
- A habit that repeats 'n' times in a day (Drinking six glasses of water a day).
  - instead of cross that turns to check, increment zero to 'n' for every click, when a habit is set as repetitive in a day.
 - A time based habit (meditate for 10 mins atleast).
   - i might do 6min first day, and i slowly improve, to see my progress this would be helpful.

I know these are a lot to ask and takes time to create but just few suggestions, may be the motto is to stay simple and minimal, but these extras would be great to have.

**Comment by hiqua** [comment]
Numeric habits are coming with v2 (soon) and solve 2) and 3). 1) is done by negating the habit "Did I avoid smoking today?".
